db.js

```diff
@@ -256,6 +262,41 @@ exports.init = function (sbot, dir, config) {
     indexes[index.name] = index
   }
 
+  function updateIndexes() {
+    const start = Date.now()
+
+    const indexesRun = Object.values(indexes)
+    let isLive = false
+
+    function liveStream() {
+      debug('live streaming changes')
+      isLive = true
+      log.stream({ gt: indexes['base'].seq.value, live: true }).pipe({
+        paused: false,
+        write: (data) => indexesRun.forEach((x) => x.onData(data, isLive)),
+      })
+    }
+
+    const lowestSeq = Math.min(
+      ...Object.values(indexes).map((x) => x.seq.value)
+    )
+    debug(`lowest seq for all indexes ${lowestSeq}`)
+
+    log.stream({ gt: lowestSeq }).pipe({
+      paused: false,
+      write: (data) => indexesRun.forEach((x) => x.onData(data, isLive)),
+      end: () => {
+        const tasks = []
+        indexesRun.forEach((index) => {
+          tasks.push(promisify(index.writeBatch)())
+        })
```

Comment:
It would be better to avoid mutation if we can make the code simpler:
```js
const tasks = indexesRun.map(index => promisify(index.writeBatch)())
Promise.all(tasks).then(liveStream)
```